### PR TITLE
[XLA] Add option to tfcompile HLO proto

### PR DESCRIPTION
**Pull request by rjeli**

### CODE REVIEW

File: tensorflow/compiler/aot/tfcompile_main.cc [hunk 1]
```diff
@@ -81,13 +82,21 @@ Status Main(const MainFlags& flags) {
   }
 
   // Read and initialize the graph.
-  if (flags.graph.empty()) {
-    return errors::InvalidArgument("Must specify --graph");
+  if (flags.graph.empty() && flags.hlo_module.empty()) {
+    return errors::InvalidArgument("Must specify either --graph or --hlo_module");
```
Comment (sanjoy) [inline tensorflow/compiler/aot/tfcompile_main.cc]: Also check that the user didn't specify both.

File: tensorflow/compiler/aot/flags.cc [hunk 2]
```diff
@@ -24,6 +24,8 @@ void AppendMainFlags(std::vector<Flag>* flag_list, MainFlags* flags) {
        "Input GraphDef file.  If the file ends in '.pbtxt' it is expected to "
        "be in the human-readable proto text format, otherwise it is expected "
        "to be in the proto binary format."},
+      {"hlo_module", &flags->hlo_module,
```
Comment (sanjoy) [inline tensorflow/compiler/aot/flags.cc]: State that the user may supply one of `--graph` or `--hlo_module`.

Comment (sanjoy) [reply tensorflow/compiler/aot/flags.cc]: Please also add a warning that the HLO proto is *not* a backwards / forwards compatible format and it is not okay to e.g. expect an HLO module generated today will continue to work in perpetuity with tfcompile.

**Review by sanjoy — CHANGES_REQUESTED** [review]

**Comment by rjeli** [comment]
Ok, thanks for review. My first TF PR so want to make sure I have the style/conventions right 🙂

File: tensorflow/compiler/aot/tfcompile_main.cc [hunk 3]
```diff
@@ -80,14 +81,24 @@ Status Main(const MainFlags& flags) {
     return Status::OK();
   }
 
-  // Read and initialize the graph.
-  if (flags.graph.empty()) {
-    return errors::InvalidArgument("Must specify --graph");
+  if ((flags.graph.empty() && flags.hlo_module.empty())
+   || (!flags.graph.empty() && !flags.hlo_module.empty())) {
```
Comment (rjeli) [inline tensorflow/compiler/aot/tfcompile_main.cc]: This is just an XNOR (`flags.graph.empty() == flags.hlo_module.empty()`), but I assume clarity is more important

**Review by sanjoy — APPROVED** [review]
lgtm

**Comment by sanjoy** [comment]
This breaks some internal tests.  You can reproduce the error `bazel test //tensorflow/compiler/aot/tests:all_tests -c opt`, we don't test these by default since that would end up building LLVM twice, once for the host build and once for the target build.

Right now `//tensorflow/compiler/aot/tests:all_tests` is broken for some unrelated reasons, but I have a fix in flight that should land soon.

**Comment by gbaned** [comment]
@sanjoy  Any update on this PR, please.

**Comment by sanjoy** [comment]
> Right now `//tensorflow/compiler/aot/tests:all_tests` is broken for some unrelated reasons, but I have a fix in flight that should land soon.

This is fixed now so @rjeli should be able to run the AOT tests with this patch and fix them.

**Review by sanjoy — CHANGES_REQUESTED** [review]
The AOT unit tests need to be fixed, as stated in the conversation.

**Comment by gbaned** [comment]
@rjeli Could you please address the reviewer comments. Thanks!

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**